core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -18,280 +18,416 @@
-      var caName = customizationArgSpecs[i].name;
+  constructor(
+    private ngbActiveModal: NgbActiveModal,
+    private externalRteSaveService: ExternalRteSaveService,
+    private alertsService: AlertsService,
+    private fb: FormBuilder,
+    private assetsBackendApiService: AssetsBackendApiService,
+    private contextService: ContextService,
+    private focusManagerService: FocusManagerService,
+    private imageLocalStorageService: ImageLocalStorageService,
+    private imageUploadHelperService: ImageUploadHelperService
+  ) {}
+
+  ngOnInit(): void {
+    this.focusManagerService.setFocus('tmpFocusPoint');
+    for (let i = 0; i < this.customizationArgSpecs.length; i++) {
+      const caName = this.customizationArgSpecs[i].name;
       if (caName === 'math_content') {
-        $scope.currentRteIsMathExpressionEditor = true;
-        var mathValueDict = {
+        this.currentRteIsMathExpressionEditor = true;
+        const mathValueDict = {
           name: caName,
-          value: (
-            attrsCustomizationArgsDict.hasOwnProperty(caName) ?
-              angular.copy(attrsCustomizationArgsDict[caName]) :
-              customizationArgSpecs[i].default_value)
-        };
+          value: this.attrsCustomizationArgsDict.hasOwnProperty(caName) ?
+            typedCloneDeep(this.attrsCustomizationArgsDict[caName]) :
+            this.customizationArgSpecs[i].default_value,
+        } as Extract<
+          CustomizationArgsNameAndValueArray[number],
+          { name: typeof caName }
+        >;
         // If the component being created or edited is math rich text component,
         // we need to pass this extra attribute svgFile to the math RTE editor.
         // The math RTE editor will auto-generate the svgFile based on the
         // rawLatex value and then this file can be saved to the backend when
         // the user clicks on the save button.
         mathValueDict.value.svgFile = null;
         mathValueDict.value.mathExpressionSvgIsBeingProcessed = false;
-        $scope.tmpCustomizationArgs.push(mathValueDict);
+        (
+          this.tmpCustomizationArgs as Extract<
+            CustomizationArgsNameAndValueArray[number],
+            { name: typeof caName }
+          >[]
+        ).push(mathValueDict);
       } else {
-        $scope.tmpCustomizationArgs.push({
+        const t = {
           name: caName,
-          value: (
-            attrsCustomizationArgsDict.hasOwnProperty(caName) ?
-              angular.copy(attrsCustomizationArgsDict[caName]) :
-              customizationArgSpecs[i].default_value)
-        });
+          value: this.attrsCustomizationArgsDict.hasOwnProperty(caName) ?
+            angular.copy(this.attrsCustomizationArgsDict[caName]) :
+            this.customizationArgSpecs[i].default_value,
+        } as Extract<
+          CustomizationArgsNameAndValueArray[number],
+          { name: typeof caName }
+        >;
+        (
```

Comment:
Ditto (re "this feels clumsy").

Reply:
Thanks, see above for what to do here (comment + TODO).

Reply:
Done